src/main/java/nus/climods/storage/acadyearmodulelist/JsonSerializableAcadYearModuleList.java

```diff
@@ -0,0 +1,39 @@
+package nus.climods.storage.acadyearmodulelist;
+
+import java.util.List;
+
+import org.openapitools.client.model.ModuleCondensed;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonRootName;
+
+import nus.climods.commons.exceptions.IllegalValueException;
+import nus.climods.model.AcadYearModuleList;
+
+/**
+ * An Immutable acadYearModuleList that is serializable to JSON format.
+ */
+@JsonRootName(value = "modules")
+class JsonSerializableAcadYearModuleList {
+
+    private final List<ModuleCondensed> modules;
+
+    /**
+     * Constructs a {@code JsonSerializableAcadYearModuleList} with the given modules.
+     */
+    @JsonCreator
+    public JsonSerializableAcadYearModuleList(@JsonProperty("persons") List<ModuleCondensed> modules) {
```

Comment:
```suggestion
    public JsonSerializableAcadYearModuleList(@JsonProperty("modules") List<ModuleCondensed> modules) {
```

`@JsonProperty` is for helping deserialising of json.  But since the name used is `modules`, I think the "persons" is ignored.  I missed out changing "persons" to "modules". Thanks for pointing it out!